src/popup/router/components/NetworkRow.vue

```diff
@@ -0,0 +1,103 @@
+<template>
+  <div class="network-row">
+    <div
+      class="network-row-wrapper"
+      :class="{ 'row-disabled': !isActive }"
+    >
+      <RadioButton
+        :value="isActive"
+        type="radio"
+        name="activeNetwork"
+        @input="selectNetwork(network.name)"
+      >
+        <p
+          class="name"
+          data-cy="network-name"
+        >
+          {{ network.name }}
+        </p>
+      </RadioButton>
+      <p
+        class="url shifted"
+        data-cy="network-url"
+      >
+        <span class="url-label">{{ $t('pages.network.url') }}</span> {{ network.url }}
+      </p>
+      <p
+        class="url shifted"
+        data-cy="network-middleware"
+      >
+        <span class="url-label">
+          {{ $t('pages.network.middleware') }}
+        </span> {{ network.middlewareUrl }}
+      </p>
+    </div>
+    <NodeConnectionStatus
+      v-if="isActive"
+      is-relative
+    />
+  </div>
+</template>
+
+<script>
+import { mapActions, mapGetters } from 'vuex';
+import NodeConnectionStatus from './NodeConnectionStatus.vue';
+import RadioButton from './RadioButton.vue';
+
+export default {
+  components: {
+    RadioButton,
+    NodeConnectionStatus,
+  },
+  props: {
+    network: {
+      type: Object,
+      required: true,
+    },
+  },
+  computed: {
+    ...mapGetters(['activeNetwork']),
+    isActive() {
+      return this.network === this.activeNetwork;
+    },
+  },
+  methods: mapActions(['selectNetwork']),
+};
+</script>
+
+<style lang="scss" scoped>
+@use '../../../styles/variables';
+@use '../../../styles/typography';
+
+.network-row {
```

Comment:
We are using a nesting approach with selectors.
You shall consider using it. https://www.w3.org/TR/css-nesting-1/
Or to reconsider this approach with your own arguments and create an issue about it.